ckanext/datastore/logic/action.py

```diff
@@ -248,8 +248,13 @@ def datastore_run_triggers(context, data_dict):
     '''
     res_id = data_dict['resource_id']
     p.toolkit.check_access('datastore_run_triggers', context, data_dict)
-    backend = DatastoreBackend.get_active_backend()
-    connection = backend._get_write_engine().connect()
+
+    # (canada fork only): use contextual connection so that the temp
+    #                     ds user table persists in the connection.
+    connection = context.get('connection', None)
```

Comment:
yeah, suggesting to change the name on both sides

Reply:
Ah okay. The context manager uses `connection` however because it is used in another place and that `connection` context is set by the DataStore Postgres methods: https://github.com/open-data/ckanext-canada/blob/master/ckanext/canada/plugins.py#L388

I guess I could modify our context manager to check for `connection` or `datastore_write_connection`??

Reply:
If that name is used already I guess we're stuck with it. Carry on.